### ISSUE TITLE
Right-click context menu from icon in macOS dock shows different recently opened files/folders from File->Open Recent

### ISSUE THREAD

**Issue by kumarbhushan** [microsoft/vscode#72318]
- VSCode Version: Code 1.33.0
- OS Version:macOs mojave 10.14.3

Steps to Reproduce:

1.Launch VS Code, then choose a folder to open
2.Close the app
3.Right-click on dock icon; more than likely this folder will not be available as a shortcut (nor will it appear if click "Show Recents")

I would expect that the most-recently used files and folders would appear in this context menu as they appear in the File->Open Recent menu

Reproduces without extensions: N/A

![download](https://user-images.githubusercontent.com/10768018/56114171-b1cc0780-5f7d-11e9-8462-66d333b95f5d.png)

**Comment by bpasero** [comment]
No, the list in the dock is filtered by the entries existing on disk so there can be differences. I am not sure if the welcome experience is doing any other filtering?

**Comment by chrmarti** [comment]
The Welcome page shows the recent list of folders without the current folder.

**Comment by peteszym** [comment]
@kumarbhushan , I noticed this happening as well for me, and it's persisted through several VSCode and Mac OS updates. Basically, the contextual menu in the dock shows a dated list of recent folders, which essentially makes it less useful as a switching point for projects.

I seem to have managed to fix the problem by clearing the recently opened items and starting fresh. Just go to File -> Open Recent -> Clear Recently Open. After I did that and opened a few projects up, the contextual menu and welcome page seem to display the same results, at least for now.
